python/Ganga/new_tests/GPI/TestSubjob.py

```diff
@@ -0,0 +1,56 @@
+from GangaUnitTest import GangaUnitTest
+
+class TestSubjobs(GangaUnitTest):
+
+    def setUp(self):
+        """Make sure that the Job object isn't destroyed between tests"""
+        super(TestSubjobs, self).setUp()
+        from Ganga.Utility.Config import setConfigOption
+        setConfigOption('TestingFramework', 'AutoCleanup', 'False')
+
+    def testLargeJobSubmission(self):
+        """
+        Create lots of subjobs and submit it
+        """
+        from Ganga.GPI import Job, GenericSplitter, Local
+        j = Job()
+        j.application.exe = "sleep"
+        j.splitter = GenericSplitter()
+        j.splitter.attribute = 'application.args'
+        j.splitter.values = [ ['400'] for i in range(0, 100) ]
+        j.backend = Local()
+        j.submit()
+
+        self.assertEqual( len(j.subjobs), 100 )
+
+    def testSetParentOnLoad(self):
+        """
+        Test that the parents are set correctly on load
+        """
+        from Ganga.GPI import jobs, queues, Executable, Local
+        from Ganga.GPIDev.Base.Proxy import isType
+
+        def flush_full_job():
+            j = jobs(0)
+            j._impl._setDirty()
```

Comment:
What would be the preference? I'm not wedded to this, just wanted make sure the job was considered dirty as otherwise the flush wouldn't do anything :smile: